Connecting backend

StinaSoile opened the pull request:
Frontend hakee kuvat ja muut sarjakuvien tiedot backendista.
Backend toisessa repositoryssa ja pyörii osoitteessa https://matikkasarjis-backend.onrender.com.
package.jsonissa on vaihtoehtoiset scriptit lokaalin ja globaalin backendin käytölle kehitysvaiheessa.

StinaSoile committed:
Sarjakuvat haetaan nyt backendista:
- Page[]-listat, joissa tiedot sarjakuvista ja sivuihin liittyvät kysymykset
- kuvat

StinaSoile committed:
Poistettu assets-tiedosto
poistettu kommentteja

StinaSoile committed:
production ja development modet asetettu:
npm run dev käyttää lokaalia backendiä, npm run dev-prod käyttää netissä olevaa.

sepi4 commented on src/App.tsx:
```diff
@@ -34,18 +28,11 @@ function App() {
           />
           <Route
```
this can be in loop

sepi4 replied:
something like this
```
comicNames = ["siivetonlepakko", "velhontaloudenhoitaja"]
...
comicNames.map ...
...
```

sepi4 commented on src/constants.ts:
```diff
@@ -0,0 +1,7 @@
+// export const apiBaseUrl = "https://matikkasarjis-backend.onrender.com/api";
+// export const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+export const apiBaseUrl =
+  process.env.NODE_ENV === "production"
```
maybe change to NODE_ENV to BACKEND_ENV

sepi4 replied:
if changed, need to change it in other places also, like in package.json

StinaSoile replied:
if changed, the code breaks. For some reason it doesn't get recognized.

sepi4 replied:
True. Looks like it is not available on client side browser. 
Looks like you need to use vite method of setting custom env variables. 

Here good link with examples. 
https://vitejs.dev/guide/env-and-mode

So you can set variables in terminal commands like this

![image](https://github.com/user-attachments/assets/1cada0b5-aab6-4049-8cb6-b3ccbd6c51e7)

![image](https://github.com/user-attachments/assets/699416ba-d9ea-4e61-9dec-bb4f815a3895)

Or (better (maybe)) to have separate .env files (different modes) something like in examples https://vitejs.dev/guide/env-and-mode#modes
 So `.env.dev` and `.env.prod` can be created. Each would have own variables.

sepi4 commented on src/constants.ts:
```diff
@@ -0,0 +1,7 @@
+// export const apiBaseUrl = "http://localhost:3000/api";
```
remove comments

sepi4 commented on src/components/ComicList.tsx:
```diff
@@ -1,13 +1,33 @@
     <div className="list-container center">
-
       <div
         className="list-element-made-by-sergei"
```
need to make better name

StinaSoile replied:
Done.

sepi4 commented on src/components/ComicList.tsx:
```diff
@@ -1,13 +1,33 @@
 
 const ComicList = () => {
   const navigate = useNavigate();
+  const [SiivetonLepakkoEtusivu, setSiivetonLepakkoEtusivu] = useState("");
```
I whould have `comicFrontPages` variable. 

- get all front pages
- loop front pages

Separate useState for each comic is not very good to me.

StinaSoile replied:
I don't know if I changed it for better or for worse, please look at it again!

sepi4 replied:
How about something like this? 
![image](https://github.com/user-attachments/assets/b1d51569-9d38-485f-9068-c7e93453f0e3)

sepi4 replied:
`comicInfo` thing is not so important, made it just to be able to loop over many comics front page infos.
More important `promise.all`, if you have to make many ruequests it good to make them all at once not one by one.

StinaSoile committed:
All suggestions noted except ComicList.tsx useState is still separate for all comics

StinaSoile committed:
sarjakuvien kansikuvien hakeminen ComicList-tiedostossa koodattu eri tavalla

sepi4 reviewed (changes requested):
i left comments

StinaSoile committed:
- fixed the bad code in comicList
- changed NODE_ENV to VITE_BACKEND
- added unique keys to list elements in app.tsx
- removed one console.log from comicPages.tsx
